### ISSUE TITLE
n-level undo causing exception with SortedBindingList

### ISSUE THREAD

**Issue by michaelcsikos** [MarimerLLC/csla#2302]
**Describe the bug**
In version 5.4.2, `CancelEdit()` is crashing in `SortedBindingList` in the `SourceChanged(object sender, ListChangedEventArgs e)` method handling the case `ListChangedType.ItemDeleted`. This method is being called twice with the same arguments, so when it goes to remove from the internal list it's throwing `System.ArgumentOutOfRangeException`. In 4.11.2, this was only being called once.

**Version and Platform**
CSLA version: 5.4.2
OS: Windows
Platform: WinForms

**Code that Fails**
```cs
private void Cancel(object sender, EventArgs e)
{
    bindList.RaiseListChangedEvents = false;

    Unbind(false);

    MyChildren.CancelEdit();

    bindList.RaiseListChangedEvents = true;

    BindUI();
}

public void Unbind(bool apply)
{
    IEditableObject current = null;

    if (bindList.DataSource != null && bindList.Position >= 0)
        current = bindList.Current as IEditableObject;

    if (!(bindList.DataSource is BindingSource))
    {
        if (current != null)
            bindList.DataSource = current.GetType();
        else
            bindList.DataSource = null;
    }

    if (current == null)
        return;

    if (apply)
        current.EndEdit();
    else
        current.CancelEdit(); // Index out of range in 5.4.2
}
```

**Stack Trace or Exception Detail**
```
System.ArgumentOutOfRangeException
  Message=Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  Source=mscorlib
  StackTrace:
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.RemoveAt(Int32 index)
   at Csla.SortedBindingList`1.SourceChanged(Object sender, ListChangedEventArgs e)
   at System.ComponentModel.BindingList`1.OnListChanged(ListChangedEventArgs e)
   at Csla.BusinessBindingListBase`2.RemoveItem(Int32 index)
   at System.Collections.ObjectModel.Collection`1.Remove(T item)
   at Csla.BusinessBindingListBase`2.Csla.Core.IParent.RemoveChild(IEditableBusinessObject child)
   at Csla.Core.BusinessBase.System.ComponentModel.IEditableObject.CancelEdit()
```

**Additional context**
Identical sample projects for 4.11.2 which works, and 5.4.2 which throws:
[CslaSortedBindingListTest.zip](https://github.com/MarimerLLC/csla/files/6526146/CslaSortedBindingListTest.zip)

**Comment by michaelcsikos** [comment]
Hi @rockfordlhotka, have you had a chance to look at this yet?

**Comment by rockfordlhotka** [comment]
I have not - I'm struggling to resolve state management in server-side Blazor, which turns out to be really challenging!

What would be very helpful is if you could reproduce the issue in a unit test here: https://github.com/MarimerLLC/csla/tree/main/Source/Csla.test/SortedBindingList

**Comment by rockfordlhotka** [comment]
Oh, this was a sneaky issue that probably actually affected other scenarios as well.

When I implemented `LoadListMode` I failed to fully implement it in the `MobileList` base type, so it was only actually correct in the observable collection family of types (`BusinessListBase`, `ReadOnlyListBase`).

The new mode is fully implemented in the other family of lists based on `MobileList` now, which resolves the issue.
